src/client/entity-editor/entity-editor.tsx

```diff
@@ -68,6 +69,21 @@ const EntityEditor = (props: Props) => {
 		identifierEditorVisible,
 		onSubmit
 	} = props;
+	const currentState = (useSelector((state) => state) as any).toJS();
+	// eslint-disable-next-line consistent-return
+	const handleUrlChange = React.useCallback(() => {
+		if (!_.isEqual(currentState, props.intitialState) && !currentState.submissionSection.submitted) {
+			return 'You have some unsaved changes!';
+		}
+	}, [currentState]);
+	React.useEffect(() => {
+		if (props.intitialState.editionSection && !props.intitialState.editionSection.matchingNameEditionGroups) {
+			props.intitialState.editionSection.matchingNameEditionGroups = [];
+		}
+	}, []);
```

Comment:
This doesn't seem like the right way to ensure `matchingNameEditionGroups` exists.

You can try setting `matchingNameEditionGroups: Immutable.List([]),` here where we define the default empty state for the edition section:

https://github.com/metabrainz/bookbrainz-site/blob/master/src/client/entity-editor/edition-section/reducer.ts#L45-L52